src/it/scala/djinni/GeneratorIntegrationTest.scala

```diff
@@ -967,4 +970,35 @@ class GeneratorIntegrationTest extends IntegrationTest with GivenWhenThen {
     )
     assertFileContentEquals(idlFile, OBJC_HEADERS, objcHeaderFilenames)
   }
+
+  it(
+    "should generate @Deprecated annotations for TS from @deprecated notes in comments"
+  ) {
+    Given(
+      "an IDL-file that documents deprecation using @deprecated in comments"
+    )
+    val idlFile = "deprecation"
+
+    When("generating Wasm/TS source")
+    val tsFilenames =
+      Ts("module.ts")
+    val cmd = djinniParams(
+      idlFile,
+      cpp = false,
+      objc = false,
+      java = false,
+      python = false,
+      cWrapper = false,
+      cppCLI = false,
+      wasm = true,
+      cppOmitDefaultRecordCtor = true
+    )
+
+    djinni(cmd)
+
+    Then(
+      "the @deprecated comments are generated as @Deprecated annotations"
+    )
+    assertFileContentEquals(idlFile, TS, tsFilenames)
+  }
```

Comment:
It doesn't look like deprecation support was added for wasm, so I don't think this test is needed.

Also Im not sure why the enums had to be changed to use explicit values.

Reply:
That's true that I did nothing special for deprecation support. As I understand, though, the @deprecated tag matches the JSDoc spec ( https://www.typescriptlang.org/play#example/jsdoc-deprecated ) so I figured it was worth locking this expectation in place for generated TS. 

Re: enums - I get the impression that this was done in snapchat-djinni because it was expedient to have identical code generating enums for all languages. And it was easier for me, at least initially, to make the same assumption. But since the branches are now building for the generator and the support lib, I'd be quite happy to try to revert the behaviour to how it was before.